folium/folium.py

```diff
@@ -550,32 +553,39 @@ def choropleth(self, geo_data, data=None, columns=None, key_on=None,
                             if data_min > 0 else -1)
                 data_max = (data_max if data_max > 0 else 0
                             if data_max < 0 else 1)
-            data_min, data_max = (1.01*data_min-0.01*data_max,
-                                  1.01*data_max-0.01*data_min)
-            nb_class = 6
-            color_domain = [data_min+i*(data_max-data_min)*1./nb_class
-                            for i in range(1+nb_class)]
+            nb_bins = 6
+            color_domain = list(np.linspace(data_min, data_max, nb_bins + 1))
         else:
             color_domain = None
 
         if color_domain and key_on is not None:
+            nb_bins = len(color_domain) - 1
             key_on = key_on[8:] if key_on.startswith('feature.') else key_on
-            color_range = color_brewer(fill_color, n=len(color_domain))
+            color_range = color_brewer(fill_color, n=nb_bins)
 
             def get_by_key(obj, key):
                 return (obj.get(key, None) if len(key.split('.')) <= 1 else
                         get_by_key(obj.get(key.split('.')[0], None),
                                    '.'.join(key.split('.')[1:])))
 
             def color_scale_fun(x):
-                idx = len(
-                    [
-                        u for u in color_domain if
-                        get_by_key(x, key_on) in color_data and
-                        u <= color_data[get_by_key(x, key_on)]
-                    ]
-                )
-                return color_range[idx-1]
+                key_of_x = get_by_key(x, key_on)
+                if key_of_x in color_data.keys():
```

Comment:
I was thinking if this could be made a bit simpler. Instead of two if-else statements, make two if-statements that return on something wrong and keep the happy path unindented. For example here return immediately if the value is missing, no comment needed.
```suggestion
                if key_of_x not in color_data.keys():
                    return nan_fill_color
```

Then set the value, do a if-statement whether to return on `np.isnan`, and do the rest.

This is fine as it is, so no problem if you leave it like this. I like to discuss the finishing touch, that's all.

Reply:
I'll apply the change in the next commit.

Reply:
Should be good now

Reply:
Looks good indeed, thanks.